src/pushsource/_impl/model/vms.py

```diff
@@ -24,6 +24,17 @@ def __repr__(self):
         return f"{cls_name}.{self.name}"
 
 
+@attr.s()
+class VMICloudInformation(object):
+    """Inform the cloud provider's name and account alias of a given push item."""
```

Comment:
Overall, I find the intended usage of it a bit unclear and would appreciate a more detailed doc string.

Specifically I find it unclear if this is meant to be used as:

1. the cloud provider which *should be used to publish the item*, or:
2. the cloud provider on which *the item currently exists*, or:
3. both

The PR cover message talks about using this for "delete", which sounds like case (2) as you're going to use this to look up the image in a given provider.  If it's indeed case (2) then this all looks good except that I request the doc string is fleshed out a bit with a sentence or two to explain the expected usage.

This is the kind of doc string I think we should have (and I'm assuming case 2 when writing this):

```
Information on the cloud provider associated with a given push item.

Cloud provider information is only available for VMIs which have previously
been published to a cloud. It may be used to locate an existing VMI for
manipulation, such as metadata updates or deletion.

This library doesn't define any specific cloud provider names or aliases.
Generally, a user of this library is expected to use the information here
to look up cloud access details from a configuration file or other source.
```

If the case is (1) or (3), as well as expanding the doc string, could we please rename the attribute to `dest_cloud_info` ?

The reason behind this is that I am concerned about the push item attributes becoming a confusing mix between attributes having the *current* state of some content, vs *desired* state.  For example: there would be an attribute named `build_info` saying where the push item *came from*, and then a similarly named `cloud_info` saying where the push item *should be pushed*.  There's no way a developer could be reasonably expected to remember which of these attributes represent the current vs desired state.

The last time this came up was `ContainerImagePushItem.dest_signing_key`, representing a signing key that an image *should be signed with* (not "is already signed with"), so the established convention is `dest_*`.

Reply:
Our goal with this new optional property is really just for the case 2.

I've adjusted the docstring accordingly, thanks a lot for the suggestion, I really appreciate it. I've set it with the same value as it correctly represent the goal of this class.